packages/block-editor/src/components/link-control/use-search-handler.js

```diff
@@ -108,19 +108,19 @@ export default function useSearchHandler(
 	allowDirectEntry,
 	withCreateSuggestion
 ) {
-	const { fetchSearchSuggestions, pageOnFront, pageForPosts } = useSelect(
+	const { fetchSearchSuggestions, usePageSettings } = useSelect(
 		( select ) => {
 			const { getSettings } = select( blockEditorStore );
-
+			const settings = getSettings();
 			return {
-				pageOnFront: getSettings().pageOnFront,
-				pageForPosts: getSettings().pageForPosts,
+				usePageSettings: settings.__experimentalUsePageSettings,
 				fetchSearchSuggestions:
-					getSettings().__experimentalFetchLinkSuggestions,
+					settings.__experimentalFetchLinkSuggestions,
 			};
 		},
 		[]
 	);
+	const { pageOnFront, pageForPosts } = usePageSettings();
```

Comment:
Should probably be name something else :)

Reply:
Note: I haven't accounted for `usePageSettings` being undefined yet.

* Either we allow this setting to be changed, in which case we'll have to check this at the component level and remount the component.
* Or we the setting can never be changed: in this case we passed a stable function ref, or it should always stay undefined. In that case we can ignore that rule for hooks.

Reply:
In other words: should the block setting hook be much like importing a function from a package?